src/test/scala/mesosphere/marathon/api/v2/GroupsResourceTest.scala

```diff
@@ -432,5 +432,39 @@ class GroupsResourceTest extends AkkaUnitTest with GroupCreation with JerseyTest
         rootGroup.app(PathId("/test-group/sleep/goodnight")).shouldNot(be(empty))
       }
     }
+
+    "Allow batch creation of a top-level group with enforce role and apps" in {
+      new FixtureWithRealGroupManager(initialRoot = createRootGroup(groups = Set())) {
+        val body =
+          """
+        {
+          "groups": [
+            {
+              "apps": [
+                {
+                  "id": "goodnight",
+                  "cmd": "sleep 1",
+                  "instances": 0,
+                  "role": "prod"
+                }
+              ],
+              "id": "sleep"
+            }
+          ],
+          "id": "/prod",
+          "enforceRole": true
+        }"""
+        f.service.deploy(any, any).returns(Future(Done))
+
+        val response = asyncRequest { r =>
+          groupsResource.createWithPath("", false, body.getBytes, auth.request, r)
+        }
+        response.getStatus shouldBe 201
+
+        val rootGroup = groupManager.rootGroup()
+        groupPaths(rootGroup) shouldBe Set("/", "/prod", "/prod/sleep")
+        rootGroup.app(PathId("/prod/sleep/goodnight")).value.role should be("prod")
```

Comment:
What happens if the role field isn't specified above? Is this the bug we discussed?

Reply:
Yes. I added a second test. So, the validation has not bug because there is none for apps. However, during the normalization we get `*` instead of `prod` as the app's role.

Reply:
Will you please file a JIRA issue regarding the state of validation so we can discuss the ideal path forward there? Lets add it to the current quota epic?